### ISSUE TITLE
Docker containers cannot access own services via host address

### ISSUE THREAD

**Issue by karelorigin** [hashicorp/nomad#15981]
### Nomad version

Nomad v1.4.3 (f464aca721d222ae9c1f3df643b3c3aaa20e2da7)

### Operating system and Environment details

```
Linux 5.4.0-132-generic #148-Ubuntu SMP Mon Oct 17 16:02:06 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```

### Issue

A docker container cannot access its own service via its host IP. This is problematic for services that need to be able to resolve themselves. In my case, this was [RabbitMQ](https://www.rabbitmq.com/clustering.html#cluster-formation-requirements:~:text=Therefore%20every%20cluster%20member%20must%20be%20able%20to%20resolve%20hostnames%20of%20every%20other%20cluster%20member%2C%20its%20own%20hostname%2C%20as%20well%20as%20machines%20on%20which%20command%20line%20tools%20such%20as%20rabbitmqctl%20might%20be%20used.).

### Reproduction steps

1. Run the job file. 
2. Execute: `nomad alloc exec <alloc> '/bin/sh' -c 'curl "$NOMAD_HOST_ADDR_http" -v'`

#### Expected Result

"Hello World!"

#### Actual Result

```
Trying 10.18.234.39:28949...
```

### Job file

```hcl
job "http-echo" {
  datacenters = ["dc1"]

  group "echo" {
    count = 1

    network {
      mode = "bridge"

      port "http" {
        to = 8080
      }
    }

    task "server" {
      driver = "docker"

      config {
        image = "hashicorp/http-echo:alpine"

        args = [
          "-listen", ":8080", "-text", "Hello World!",
        ]
      }
    }
  }
}
```

**Comment by karelorigin** [comment]
I would assume that if server B can access a service on server A by its host IP, that server A could too via the same address. Hopefully that makes sense.

**Comment by tgross** [comment]
@karelorigin the missing configuration option here is actually bridge "hairpin mode", which was merged into Nomad in https://github.com/hashicorp/nomad/pull/15961. That shipped in Nomad 1.5.0-beta.1 and will ship in the upcoming GA for Nomad 1.5.0.

You can see this if you try to run your http-echo job and then run another job that tries to reach it over the bridge. Here's the running http-echo job:

```
$ nomad alloc status ec51
ID                  = ec51e00d-1d21-c55e-5028-3cd445181ff1
Eval ID             = 5e549945
Name                = http-echo.echo[0]
...

Allocation Addresses (mode = "bridge"):
Label  Dynamic  Address
*http  yes      192.168.1.90:20236 -> 8080
```

Then I run an allocation for a completely different job, and exec into that:

```
$ nomad alloc exec 244ca39a curl 192.168.1.90:20236
Hello World!
```


I'm going to close this issue as fixed in 1.5.0.
